Engine/Authentication/AuthenticationSettings.cs

```diff
@@ -93,6 +104,8 @@ void PrepareRequest(HttpRequestMessage request, string domain, CancellationToken
             }
         }
 
+        private static string userAgent = null;
```

Comment:
I don't think we can cache the userAgent. If two plugins both request a client, we should be able to detect which plugins made which requests